utopia-remix/app/util/use-sort-compare-project.tsx

```diff
@@ -0,0 +1,45 @@
+import moment from 'moment'
+import React from 'react'
+import { useProjectsStore } from '../store'
+import { ProjectWithoutContent } from '../types'
+import { assertNever } from './assertNever'
+
+export function useSortCompareProject() {
+  const sortCriteria = useProjectsStore((store) => store.sortCriteria)
+  const sortAscending = useProjectsStore((store) => store.sortAscending)
+
+  return React.useCallback(
+    (a: ProjectWithoutContent, b: ProjectWithoutContent): number => {
+      switch (sortCriteria) {
+        case 'title':
+          return sortAscending ? a.title.localeCompare(b.title) : b.title.localeCompare(a.title)
+        case 'dateCreated':
+          return sortAscending
+            ? moment(a.created_at).unix() - moment(b.created_at).unix()
```

Comment:
Is there a reason to use a lib for this? Date.getTime() is enough for this, isn't it?

Reply:
Yep, but those are not dates – they are strings, Prisma is lying unfortunately. So instead of doing optimistic parsing of whatever strings come out of the ORM, I used `moment` to be sure the parsing is reliable